### PR TITLE
H-133: Fix circular import for `hash-graph-types` package

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

The GraphAPIProtocol type is defined in __init__ after the schemas are imported but the schema requires the protocol so it results in a circular dependency. By conditionally import it this will solve running the app.

### :ship: Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### :scroll: Does this require a change to the docs?

The changes in this PR:

- [x] is internal and does not require a docs change

### :spider_web: Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] does not affect the execution graph

### CODE REVIEW

File: apps/hash-ai-worker-py/poetry.lock [hunk 1]
```diff
@@ -418,9 +418,7 @@ version = "0.0.0"
 description = "Type definitions for the type system and entity validation based on schemas"
 optional = false
 python-versions = ">=3.11,<4.0"
-files = [
-    {file = "hash_graph_types-0.0.0-py3-none-any.whl", hash = "sha256:c03c391cf843e36bc392c812fb9924286377c523ccd294b6d3eeff08119c2e42"},
-]
+files = []
```
Comment (indietyp) [inline apps/hash-ai-worker-py/poetry.lock:421]: Why is that here? I assume you forgot to lock again?

**Review by indietyp — CHANGES_REQUESTED** [review]
Small thing otherwise LGTM

(Python circular dependency hell here we go^^)

File: apps/hash-ai-worker-py/poetry.lock (hunk 1 above)
Comment (TimDiekmann) [reply apps/hash-ai-worker-py/poetry.lock:421]: Because this way it's working without messing around with the lock file. 🤷🏻‍♂️ 
I will most likely remove the wheel, anyway as it's super annoying to deal with.

**Review by indietyp — APPROVED** [review]
